### PR TITLE
added python3 emulation code + six + fixed up print calls

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

File: lib/matplotlib/style/core.py [hunk 1]
```diff
@@ -54,8 +60,8 @@ def use(name):
         else:
             try:
                 rc = rc_params_from_file(style, use_default_template=False)
-                print rc
-                print type(rc)
+                print(rc)
+                print(type(rc))
```
Comment (tonysyu) [inline lib/matplotlib/style/core.py]: I think these are actually debugging statements that crept into my commit to fix a broken test. Could you remove these?

File: lib/matplotlib/style/core.py [hunk 2]
```diff
@@ -1,3 +1,9 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import six
+from six.moves import reduce, xrange, zip
```
Comment (tonysyu) [inline lib/matplotlib/style/core.py]: Are these imports from `six` necessary? The module and the imported functions don't appear to be used anywhere.

**Comment by tonysyu** [comment]
Thanks for the PR! Sorry, I missed the original notification and didn't see this until you mentioned it on the main PR.

File: lib/matplotlib/style/core.py (hunk 2 above)
Comment (tacaswell) [inline lib/matplotlib/style/core.py]: iirc, every file gets `import six`, but I removed the `from ..`

File: lib/matplotlib/style/core.py (hunk 1 above)
Comment (tacaswell) [inline lib/matplotlib/style/core.py]: done

**Comment by tonysyu** [comment]
Thanks!

**Pull request merged**